mythril/analysis/security.py

```diff
@@ -45,42 +51,57 @@ def get_detection_module_hooks(modules, hook_type="pre"):
     return dict(hook_dict)
 
 
-def get_detection_modules(entrypoint, include_modules=()):
+def get_detection_modules(entrypoint, include_modules=(), custom_modules_directory=""):
     """
 
     :param entrypoint:
     :param include_modules:
     :return:
     """
-
     module = importlib.import_module("mythril.analysis.modules.base")
     module.log.setLevel(log.level)
 
     include_modules = list(include_modules)
 
     _modules = []
 
-    if not include_modules:
-        for loader, module_name, _ in pkgutil.walk_packages(modules.__path__):
+    for loader, module_name, _ in pkgutil.walk_packages(modules.__path__):
+        if include_modules and module_name not in include_modules:
+            continue
+
+        if module_name != "base":
+            module = importlib.import_module("mythril.analysis.modules." + module_name)
+            module.log.setLevel(log.level)
+            if module.detector.entrypoint == entrypoint:
+                _modules.append(module)
+    if custom_modules_directory:
+        custom_modules = [os.path.abspath(custom_modules_directory)]
+        sys.path.append(custom_modules_directory)
```

Comment:
It is probably worth checking if the custom_modules_directory has already been appended to the python path, as this method will be called multiple times.